x-pack/plugins/apm/public/components/alerting/rule_types/transaction_error_rate_rule_type/index.tsx

```diff
@@ -102,7 +106,8 @@ export function TransactionErrorRateRuleType(props: Props) {
       onChange={(value) => {
         if (value !== params.serviceName) {
           setRuleParams('serviceName', value);
-          setRuleParams('transactionType', '');
+          setRuleParams('transactionType', undefined);
```

Comment:
was added here: https://github.com/elastic/kibana/commit/ad75d900c97ca602c4d30429bc677d83b9d7ca46 to reset the values when changing the service name but imo sending an empty string can lead to misleading results. 

As for updating an existing rule, it will remove `transactionType` param from the list and it won't filter for the `transaction.type`

However, the **biggest impact** seems to be in the rule executor. Existing rules may have filter fields with empty strings, and [filtering](https://github.com/kpatticha/kibana/blob/fce4ef8168429645a01434e19b0feaefba1a4f02/x-pack/plugins/apm/server/routes/alerts/rule_types/transaction_duration/register_transaction_duration_rule_type.ts#L150-L154) these strings would yield no results. To address this issue, we need to pass `queryEmptyString` as false when using termQuery for these fields.


For example, 
```
...termQuery(TRANSACTION_TYPE, ruleParams.transactionType, {
    queryEmptyString: false,
 }),
```

Fortunately, this issue has been fixed in [v8.6](https://github.com/elastic/kibana/pull/143881/commits/f3bb573e375388193749874654c37360cc792371), but there was a regression in my [previous PR](https://github.com/elastic/kibana/pull/154241/commits/fce4ef8168429645a01434e19b0feaefba1a4f02) that caused the issue to resurface.

The tests we have in place were not sufficient to detect the regression. 😢 

**Next**
I need to revert the change that removed the queryEmptyString parameter and ensure that empty strings are not filtered out

Reply:
> but imo sending an empty string can lead to misleading results.

Agree, in most cases `undefined` is better than `""` we just need to make sure that old rules with an empty string still works.

Reply:
> The tests we have in place were not sufficient to detect the regression. 😢

Can you fix that? :)

Reply:
[Revert the change ](https://github.com/elastic/kibana/pull/155405/commits/6084cfd3ab651bee5028899fbdb987c9aad526de)

I'll start working on the tests